backend/routes.js

```diff
@@ -467,16 +483,39 @@ route.POST('/file', {
 route.GET('/file/{hash}', {}, async function (request, h) {
   const { hash } = request.params
 
-  if (hash.startsWith('_private')) {
+  if (!hash || hash.startsWith('_private')) {
     return Boom.notFound()
   }
 
   const blobOrString = await sbp('chelonia.db/get', `any:${hash}`)
   if (!blobOrString) {
     return notFoundNoCache(h)
   }
-  return h.response(blobOrString).code(200).etag(hash)
+  let type = 'application/octet-stream'
+
+  if (CONTRACT_DATA_REGEX.test(hash)) {
+    type = 'application/vnd.shelter.contractdata+json'
+  } else if (CONTRACT_MANIFEST_REGEX.test(hash)) {
+    type = 'application/vnd.shelter.contractmanifest+json'
+  } else if (CONTRACT_SOURCE_REGEX.test(hash)) {
+    type = 'application/vnd.shelter.contracttext'
+  } else if (FILE_MANIFEST_REGEX.test(hash)) {
+    type = 'application/vnd.shelter.filemanifest+json'
+  } else if (FILE_CHUNK_REGEX.test(hash)) {
+    type = 'application/vnd.shelter.filechunk+octet-stream'
+  } else if (hash.startsWith('name=')) {
+    type = 'text/plain'
+  }
```

Comment:
Not saying you need to change this code, but have you considered something like this instead?

```js
if (hash.startsWith('name=') {
  type = 'text/plain'
} else {
  type = {
    [multicodes.SHELTER_CONTRACT_MANIFEST]: 'application/vnd.shelter.contractmanifest+json',
    [multicodes.SHELTER_CONTRACT_TEXT]: 'application/vnd.shelter.contracttext',
    // ...
  }[multicodes.extract(hash)] || 'application/octet-stream'
}
```